### PR TITLE
Initial Commit for unregistered user PMs client-side block option

### PR DESCRIPTION
**Pull request by MustyMustelidae**

### CODE REVIEW

**Comment by Daenyth** [comment]
Is this using the same setting as ignoring chat from unreg? I think it makes sense to be two options

**Comment by MustyMustelidae** [comment]
If anyone can give me a quick rundown of how you guys chose to implement localization (I noticed you guys are using TR, but any conventions you have for the actual localization files) I can add the setting.

**Comment by Daenyth** [comment]
I gather that other people have somewhat strong feelings that it should be one setting and not two, so it's probably fine as is - could people chime in on which they prefer?

**Comment by MassCraziness** [comment]
One setting seems all that is necessary. Generally speaking, people want to just block unregistered users, not just in one area or another.

**Comment by woogerboy21** [comment]
I vote 1 settings.  And from my understanding of the people i've asked they only want 1 setting to completely ignore them.

**Comment by Daenyth** [comment]
I seem to be the minority, so let's go with one setting. If people dislike it, it shouldn't be hard to expand later.

**Comment by Daenyth** [comment]
@ZeldaZach would you test this and then merge when ready?

**Comment by tooomm** [comment]
I disagree here completely!
I don't see the advantage of 1 setting over 2 at all. If there are players who still want to follow **all** chat and don't want to discriminate people for not being registered generally, there is just no way they can do that. But they still should be able to shut down personal harassment via pm because they won this one game and met a unfriendly unregistered user there...

I already see people screaming because they don't get answers of their (partially unseen, because some users on older clients will see their messages in chat) questions and people complain about incomplete chat log etc.pp. (they can't follow a conversation between unregisterd + registered users! they end up just seeing a few scraps of the conversation). Just because many people might block them just because there is no other choice, if they want to or not!

Please. Just make it one option out of two to block pm's from unregistered users.
I don't see the benefit from a cleaner interface here just because you save up one check mark field, expect you want to infantilize the users, or force to register, or...

(Sorry for me being kinda emotional here. It's very difficult for me to understand that you think like that or that you don't care about the wide-ranging influence this has to normal chatting, maybe that's because I've never had any problems with unregistered users so far...
If I'm wrong or might oversee something I'm more than happy if you tell me!)

**Comment by ZeldaZach** [comment]
After testing this feature does NOT work as expected. do not consider merge until tested again

**Comment by Daenyth** [comment]
@tooomm I agree with the general sentiment, especially about how it can be confusing to users.

The default option is to not ignore, so that shouldn't cause any problems that weren't there before.

I agree that the one extra option is probably worthwhile. Let's leave this open for a little to have people chime in and we'll make a decision

**Comment by MustyMustelidae** [comment]
@ZeldaZach I can't build changes on this machine, but looking over it there doesn't seem to be much room for things to go wrong. I assumed the receiver_name is the person who receives the messages from the tab that gets created. I've switched the sender_name and receiver_name roles in case that's backwards.

**Comment by poixen** [comment]
i think ZZ tried it logged in as a mod. Mods should not be able to unreg users en-mass.

**Comment by ZeldaZach** [comment]
So yeah, what should we do with this atm?

**Comment by Daenyth** [comment]
@MustyMustelidae Can you take a build from this commit, connect to server, put the setting in place, connect with a second client as an unreg user, message yourself, and see if it works now?

File: cockatrice/src/tab_supervisor.cpp [hunk 1]
```diff
@@ -446,11 +446,27 @@ void TabSupervisor::processGameEventContainer(const GameEventContainer &cont)
...
+    ServerInfo_User otherUser;
+
+    if (twi)
+        otherUser = twi->getUserInfo();
+    else
+        otherUser.set_name(senderName.toStdString());
+
+    UserLevelFlags otherUserLevel = UserLevelFlags(otherUser.user_level());
+
+    if (settingsCache->getIgnoreUnregisteredUsers() && !otherUserLevel.testFlag(ServerInfo_User::IsRegistered))
```
Comment (Daenyth) [inline cockatrice/src/tab_supervisor.cpp]: Why don't you test that otherUserLevel IsUnregistered?

**Comment by MassCraziness** [comment]
@Daenyth it seems that otherUserLevel is the level of the person receiving the PM (in this case, the registered user receiving the PM from the Unregistered user). Adding another line and changing it to IsModerator causes PMs to be blocked even on a Moderator-level account. I assume, changing this to IsUnregistered would allow unregistered users to block their PM. I can check that, if you'd like

**Comment by Daenyth** [comment]
I'd definitely change the variable name then - `otherUser` makes me think that it's the other user - ie not yourself but the user who is sending a message to you.

**Comment by Daenyth** [comment]
Thanks for the PR - I updated the code a bit. Sending a new pr

**Pull request closed without merging**